### ISSUE TITLE
Compare collections using a delegate

### ISSUE THREAD

**Issue by bradwilson** [xunit/xunit#2692]
### Discussed in https://github.com/xunit/xunit/discussions/2683

<div type='discussions-op-text'>

<sup>Originally posted by **vc-74** February 27, 2023</sup>
`Assert` currently supports comparing collections, optionally passing an `IEqualityComparer` implementation.

If I'm not mistaken, only `IEqualityComparer.Equals` required in this case (IMO `IEqualityComparer` should have been called `IHashable` and inherit from `IEqualityComparer` which would have only `Equals` but I digress...).

The problem with that approach is that we have to build an `IEqualityComparer` even for simple cases, it would be much simpler to allow passing an inline comparer like:

```csharp
Assert.Equal(col1, col2, (expected, actual) =>
{
    // Comparison logic here
});
```</div>

**Comment by SARAVANA1501** [comment]
https://github.com/xunit/xunit/pull/2693 Started with this, Facing some problem in submodule push, tests has been pushed, will push the assertion code soon

**Comment by bradwilson** [comment]
Shipped 408d2d6bdf5b8d92a35bd0d184dacb4689006cf3
